grant-project/zcash-grant-system
# Proposal stage and comment pagination

Closes #121. Closes #191.

### What this does
- adds "more" style pagination to proposal comments
- gets all proposal stage filters working
- adds "all" option to proposal stage filter

### Notes
- paginates on 10 top level comments, so if there is a lot of nesting the pages could be long
- camping for new comments will require reloading/revisiting the proposal detail page, as the first page is cached and doesn't update after first load of comments tab

### Testing
- you may use `flask create-proposals 1` to get a proposal with 100 comments
- go check out that proposal's comments, make sure "Older Comments" button works correctly
- ensure you can add a new top-level comment (and reload, comment should appear)
- ensure you can create nested comments (and reload, comments should appear)
- ensure you can add a comment to a proposal without any comments (and reload, comment should load)

- go to proposals and test the stage filters
- try the stage filters on admin too

### Screens
![comments-pagination](https://user-images.githubusercontent.com/11430954/52907287-4b986200-3224-11e9-890b-795acc3b392f.gif)
